File: api/src/main/java/fr/huiitre/tools/modules/riot/sync/infrastructure/PostgresValorantSkinChromaSyncRepository.java
package fr.huiitre.tools.modules.riot.sync.infrastructure;

import org.springframework.jdbc.core.JdbcTemplate;

import fr.huiitre.tools.modules.riot.sync.application.ValorantSkinChromaSyncData;
import fr.huiitre.tools.modules.riot.sync.application.ValorantSkinChromaSyncRepository;

public class PostgresValorantSkinChromaSyncRepository implements ValorantSkinChromaSyncRepository {

    private final JdbcTemplate jdbcTemplate;

    public PostgresValorantSkinChromaSyncRepository(JdbcTemplate jdbcTemplate) {
        this.jdbcTemplate = jdbcTemplate;
    }

    @Override
    public void deleteAll() {
        jdbcTemplate.update("DELETE FROM tools_riot.valorant_skin_chromas");
    }

    @Override
    public void save(Long skinId, ValorantSkinChromaSyncData data) {
        final String sql = """
                INSERT INTO tools_riot.valorant_skin_chromas
                    (skin_id, asset_id, chroma_index, name, display_icon_url, full_render_url, swatch_url, streamed_video_url)
                VALUES (?, ?, ?, ?, ?, ?, ?, ?)
                """;
        jdbcTemplate.update(sql,
                skinId,
                data.getAssetId(),
                data.getChromaIndex(),
                data.getName(),
                data.getDisplayIconUrl(),
                data.getFullRenderUrl(),
                data.getSwatchUrl(),
                data.getStreamedVideoUrl());
    }
}

File: api/Modules/Riot/Sync/Infrastructure/PostgresValorantSkinChromaSyncRepository.cs
using Tools.Api.Modules.Riot.Sync.Application;
using Tools.Api.Modules.Riot.Common.Infrastructure;
using Tools.Api.Modules.Riot.Sync.Application.Ports;

namespace Tools.Api.Modules.Riot.Sync.Infrastructure;

public sealed class PostgresValorantSkinChromaSyncRepository(RiotDatabase database) : IValorantSkinChromaSyncRepository
{
    public Task DeleteAll() =>
        database.Execute("DELETE FROM tools_riot.valorant_skin_chromas");

    public Task Save(long skinId, ValorantSkinChromaSyncData data) =>
        database.Execute(
            """
            INSERT INTO tools_riot.valorant_skin_chromas
                (skin_id, asset_id, chroma_index, name, display_icon_url, full_render_url, swatch_url, streamed_video_url)
            VALUES (@SkinId, @AssetId, @ChromaIndex, @Name, @DisplayIconUrl, @FullRenderUrl, @SwatchUrl, @StreamedVideoUrl)
            """,
            new { SkinId = skinId, data.AssetId, data.ChromaIndex, data.Name, data.DisplayIconUrl, data.FullRenderUrl, data.SwatchUrl, data.StreamedVideoUrl });
}
